lib/linter/linter.js

```diff
@@ -1271,11 +1271,13 @@ class Linter {
      * Initialize the Linter.
      * @param {Object} [config] the config object
      * @param {string} [config.cwd] path to a directory that should be considered as the current working directory, can be undefined.
+     * @param {Array<string>} [config.flags] the feature flags to enable.
      * @param {"flat"|"eslintrc"} [config.configType="flat"] the type of config used.
      */
-    constructor({ cwd, configType = "flat" } = {}) {
+    constructor({ cwd, configType = "flat", flags = [] } = {}) {
         internalSlotsMap.set(this, {
             cwd: normalizeCwd(cwd),
+            flags,
```

Comment:
Same as above, shall we create a defensive copy of the input `flags`?

```suggestion
            flags: [...flags],
```